timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java

```diff
@@ -423,6 +423,55 @@ public void nearbyCommitsDoNotAffectResultsPresentInCache() {
                 .doesNotThrowAnyException();
     }
 
+    @Test
+    public void lateAbortingTransactionDoesNotFlushValuesToCentralCache() {
+        txnManager.runTaskThrowOnConflict(txn -> {
+            txn.put(TABLE_REF, ImmutableMap.of(CELL_1, DATA_1, CELL_2, DATA_2, CELL_3, DATA_3));
+            return null;
+        });
+
+        awaitUnlock();
+
+        AtomicLong startTimestamp = new AtomicLong(-1L);
+        PreCommitCondition commitFailingCondition = timestamp -> {
+            if (timestamp != startTimestamp.get()) {
+                throw new RuntimeException("Transaction failed at commit time");
+            }
+        };
+        try {
+            txnManager.runTaskWithConditionThrowOnConflict(commitFailingCondition, (txn, _unused) -> {
+                startTimestamp.set(txn.getTimestamp());
+                txn.put(TABLE_REF, ImmutableMap.of(CELL_4, DATA_4));
+                txn.get(TABLE_REF, ImmutableSet.of(CELL_1, CELL_2, CELL_3));
+                return null;
+            });
```

Comment:
please insert a `fail();` after this line: we don't want the test to pass if this transaction actually commits

Reply:
Refactored to just use `assertThatThrownBy` since that's just better.